Gemfile

```diff
@@ -0,0 +1,71 @@
+source ENV['GEM_SOURCE'] || 'https://rubygems.org'
+
+def location_for(place_or_version, fake_version = nil)
+  git_url_regex = %r{\A(?<url>(https?|git)[:@][^#]*)(#(?<branch>.*))?}
+  file_url_regex = %r{\Afile:\/\/(?<path>.*)}
+
+  if place_or_version && (git_url = place_or_version.match(git_url_regex))
+    [fake_version, { git: git_url[:url], branch: git_url[:branch], require: false }].compact
+  elsif place_or_version && (file_url = place_or_version.match(file_url_regex))
+    ['>= 0', { path: File.expand_path(file_url[:path]), require: false }]
+  else
+    [place_or_version, { require: false }]
+  end
+end
+
+ruby_version_segments = Gem::Version.new(RUBY_VERSION.dup).segments
+minor_version = ruby_version_segments[0..1].join('.')
+
+group :development do
```

Comment:
Why are these gems special ?

Reply:
This file is boilerplate emitted by the pdk (puppet development kit) for bootstrapping or converting an existing puppet mod.  Using named groups of gems is more or less a bundler best practice and its pretty common to have gems present for development that should be excluded from a production deploy, acceptance tests, etc. -- I don't think it makes any practice difference in this case as there is not a separate group for acceptance testing.

There is voodoo going on in this block to make the setup work across multiple ruby versions and on different platforms.  The goal is to pull in the correct `puppet-module-*` gem, which is essentially a meta package that pulls in a bunch of development tools.